lib/assets/javascripts/src/local-time/controller.coffee

```diff
@@ -27,8 +27,11 @@ class LocalTime.Controller
 
   processElement: (element) ->
     datetime = element.getAttribute("datetime")
-    format = element.getAttribute("data-format")
     local = element.getAttribute("data-local")
+    format = if config.format24 and element.hasAttribute("data-format24")
```

Comment:
> as we're coming from 12h only

Yeah I think this is what breaks it for me. I'd love to have symmetry with ISO and Intl.DateTimeFormat (https://github.com/basecamp/local_time/pull/135#discussion_r1236446931) but basing the conditionals and config off of `hour12` would depart from the existing assumption that this is a 12-hour-first library. If only because it was built that way initially.

This assumption is present, for example, in the convention that `data-format` should be used for 12-hour formats, whereas `data-format24` should be used for 24-hour formats.

In my mind, `data-format` is clearly the norm and `data-formatXX` departs from that norm. If hour12 is the norm, then checking if we're following the norm is strange. I'd rather check if we're departing from the norm, if that makes sense.

Reply:
Thank you for the context @josefarias. Given this I agree that this change is too big here, good call!